File: HTTPShortcuts/app/src/main/java/ch/rmy/android/http_shortcuts/utils/OnItemChosenListener.java
package ch.rmy.android.http_shortcuts.utils;

import android.view.View;
import android.widget.AdapterView;

import com.farbod.labelledspinner.LabelledSpinner;

public abstract class OnItemChosenListener implements LabelledSpinner.OnItemChosenListener {

    @Override
    public void onItemChosen(View labelledSpinner, AdapterView<?> adapterView, View itemView, int position, long id) {
        onSelectionChanged();
    }

    public abstract void onSelectionChanged();

    @Override
    public void onNothingChosen(View labelledSpinner, AdapterView<?> adapterView) {

    }

}

File: HTTPShortcuts/app/src/main/kotlin/ch/rmy/android/http_shortcuts/utils/OnItemChosenListener.kt
package ch.rmy.android.http_shortcuts.utils

import android.view.View
import android.widget.AdapterView

import com.satsuware.usefulviews.LabelledSpinner

abstract class OnItemChosenListener : LabelledSpinner.OnItemChosenListener {

    override fun onItemChosen(labelledSpinner: View?, adapterView: AdapterView<*>?, itemView: View?, position: Int, id: Long) {
        onSelectionChanged()
    }

    abstract fun onSelectionChanged()

    override fun onNothingChosen(labelledSpinner: View?, adapterView: AdapterView<*>?) {

    }

}
